From 5b39dd87c164db0e924f9c646d119495f1208740 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sun, 29 Mar 2026 23:01:03 +1100
Subject: [PATCH] dbutils: fix order of libraries for static build

---
 appl/dbutils/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/appl/dbutils/Makefile.am b/appl/dbutils/Makefile.am
index a1fc3842a0..53b6879e70 100644
--- a/appl/dbutils/Makefile.am
+++ b/appl/dbutils/Makefile.am
@@ -10,4 +10,4 @@ man_MANS = bsearch.1
 
 EXTRA_DIST = NTMakefile $(man_MANS)
 
-LDADD = $(LIB_roken) $(LIB_vers) $(LIB_heimbase)
+LDADD = $(LIB_vers) $(LIB_heimbase) $(LIB_roken)
