From 7f5c039a1500f8bc24db088b181de1940b7d2ebf Mon Sep 17 00:00:00 2001
From: KMojek <k.mojek@techsmith.com>
Date: Wed, 25 Aug 2021 10:42:14 -0400
Subject: [PATCH] Update CMakeLists.txt to build mp4v2 and mp4info

---
 CMakeLists.txt | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2981d44..67a3835 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -197,21 +197,21 @@ set(SOURCE_FILES
         src/rtphint.cpp
         src/text.cpp)
 
-#add_library(mp4v2-shared SHARED ${HEADER_FILES} ${SOURCE_FILES})
+add_library(mp4v2 SHARED ${HEADER_FILES} ${SOURCE_FILES})
 # Just expose a static lib for now
-add_library(mp4v2 STATIC ${HEADER_FILES} ${SOURCE_FILES})
+#add_library(mp4v2 STATIC ${HEADER_FILES} ${SOURCE_FILES})
 
 target_include_directories(mp4v2 PRIVATE
    ${CMAKE_CURRENT_SOURCE_DIR}
    ${CMAKE_CURRENT_SOURCE_DIR}/include
    )
 
-target_compile_definitions(mp4v2 PUBLIC MP4V2_USE_STATIC_LIB)
-
+#target_compile_definitions(mp4v2 PUBLIC MP4V2_USE_STATIC_LIB)
+target_compile_definitions(mp4v2 PRIVATE MP4V2_EXPORTS)
 
 #set(UTILITY_HEADERS
 #        util/impl.h)
-#
+
 #add_executable(mp4art ${UTILITY_HEADERS} util/mp4art.cpp)
 #target_link_libraries(mp4art mp4v2-static)
 #
@@ -225,7 +225,12 @@ target_compile_definitions(mp4v2 PUBLIC MP4V2_USE_STATIC_LIB)
 #target_link_libraries(mp4file mp4v2-static)
 #
 #add_executable(mp4info ${UTILITY_HEADERS} util/mp4info.cpp)
-#target_link_libraries(mp4info mp4v2-static)
+add_executable(mp4info util/mp4info.cpp)
+target_include_directories(mp4info PRIVATE SYSTEM
+   ${CMAKE_CURRENT_SOURCE_DIR}
+   ${CMAKE_CURRENT_SOURCE_DIR}/include
+   )
+target_link_libraries(mp4info mp4v2)
 #
 #add_executable(mp4subtitle ${UTILITY_HEADERS} util/mp4subtitle.cpp)
 #target_link_libraries(mp4subtitle mp4v2-static)
