From 7c3fb80adcb2a41f15267edcfb73200a3f3e1f78 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Fri, 2 Jan 2026 22:11:31 +0200
Subject: [PATCH 12/14] Update NEWS

---
 NEWS | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/NEWS b/NEWS
index 12390745..da461a5c 100644
--- a/NEWS
+++ b/NEWS
@@ -2,6 +2,24 @@ NEWS - list of user-visible changes between releases of GNU Libtool
 
 * Noteworthy changes in release ?.? (????-??-??) [?]
 
+** Bug fixes:
+
+  - Fix syntax error in LT_PROG_OBJC and LT_PROG_OBJCXX.
+
+  - Separate Objective C and C++ cache check for proper tagging support.
+
+  - Fix in darwin to support values with spaces.
+
+  - Limit the length of DLL name to 8.3 correctly to avoid corrupting a
+    generated DLL on OS/2.
+
+  - Remove unused variable on OS/2, which could cause issues with static
+    library generation if defined.
+
+** Changes in supported systems or compilers:
+
+  - Add support for SlimCC compiler.
+
 
 * Noteworthy changes in release 2.6.0 (2025-09-18) [alpha]
 
-- 
2.53.0

