From 381d0a0907bae057f12b904db005b362aa828cff Mon Sep 17 00:00:00 2001
From: npzacs <npzacs@gmail.com>
Date: Fri, 4 Mar 2022 10:00:46 +0000
Subject: [PATCH 04/12] Fix leak

---
 src/libaacs/aacs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libaacs/aacs.c b/src/libaacs/aacs.c
index 2dedefe..221965c 100644
--- a/src/libaacs/aacs.c
+++ b/src/libaacs/aacs.c
@@ -661,6 +661,7 @@ static int _calc_mk(AACS *aacs, uint8_t *mk, pk_list *pkl, dk_list *dkl)
 
     if (aacs->cc && aacs->cc->aacs2) {
         /* note: this does not detect HDMV only discs */
+        mkb_close(mkb);
         BD_DEBUG(DBG_AACS | DBG_CRIT, "Error calculating media key: disc is using AACS2\n");
         return AACS_ERROR_UNSUPPORTED_DISC;
     }
-- 
2.43.0

