diff -ur a/src/libs/zbxalgo/Makefile.in b/src/libs/zbxalgo/Makefile.in
--- a/src/libs/zbxalgo/Makefile.in	2021-12-23 08:49:11.000000000 +0000
+++ b/src/libs/zbxalgo/Makefile.in	2022-01-24 01:47:15.364730229 +0000
@@ -141,7 +141,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/algodefs.Po \
diff -ur a/src/libs/zbxcommon/Makefile.in b/src/libs/zbxcommon/Makefile.in
--- a/src/libs/zbxcommon/Makefile.in	2021-12-23 08:49:11.000000000 +0000
+++ b/src/libs/zbxcommon/Makefile.in	2022-01-24 01:45:14.948372876 +0000
@@ -141,7 +141,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/libzbxcommon_a-alias.Po \
diff -ur a/src/libs/zbxcomms/Makefile.in b/src/libs/zbxcomms/Makefile.in
--- a/src/libs/zbxcomms/Makefile.in	2021-12-23 08:49:11.000000000 +0000
+++ b/src/libs/zbxcomms/Makefile.in	2022-01-24 01:57:09.166121130 +0000
@@ -133,7 +133,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/comms.Po ./$(DEPDIR)/telnet.Po
diff -ur a/src/libs/zbxcompress/Makefile.in b/src/libs/zbxcompress/Makefile.in
--- a/src/libs/zbxcompress/Makefile.in	2021-12-23 08:49:11.000000000 +0000
+++ b/src/libs/zbxcompress/Makefile.in	2022-01-24 02:00:02.202297571 +0000
@@ -133,7 +133,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/libzbxcompress_a-compress.Po
diff -ur a/src/libs/zbxconf/Makefile.in b/src/libs/zbxconf/Makefile.in
--- a/src/libs/zbxconf/Makefile.in	2021-12-23 08:49:11.000000000 +0000
+++ b/src/libs/zbxconf/Makefile.in	2022-01-24 01:48:38.534520675 +0000
@@ -133,7 +133,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/cfg.Po
diff -ur a/src/libs/zbxcrypto/Makefile.in b/src/libs/zbxcrypto/Makefile.in
--- a/src/libs/zbxcrypto/Makefile.in	2021-12-23 08:49:11.000000000 +0000
+++ b/src/libs/zbxcrypto/Makefile.in	2022-01-24 01:44:27.748496785 +0000
@@ -139,7 +139,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/libzbxcrypto_a-aes.Po \
diff -ur a/src/libs/zbxexec/Makefile.in b/src/libs/zbxexec/Makefile.in
--- a/src/libs/zbxexec/Makefile.in	2021-12-23 08:49:11.000000000 +0000
+++ b/src/libs/zbxexec/Makefile.in	2022-01-24 01:58:14.335933853 +0000
@@ -133,7 +133,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/execute.Po
diff -ur a/src/libs/zbxhttp/Makefile.in b/src/libs/zbxhttp/Makefile.in
--- a/src/libs/zbxhttp/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxhttp/Makefile.in	2022-01-24 01:49:22.367704930 +0000
@@ -134,7 +134,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/http.Po ./$(DEPDIR)/punycode.Po \
diff -ur a/src/libs/zbxipcservice/Makefile.in b/src/libs/zbxipcservice/Makefile.in
--- a/src/libs/zbxipcservice/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxipcservice/Makefile.in	2022-01-24 01:59:33.055712047 +0000
@@ -134,7 +134,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/libzbxipcservice_a-ipcservice.Po
diff -ur a/src/libs/zbxjson/Makefile.in b/src/libs/zbxjson/Makefile.in
--- a/src/libs/zbxjson/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxjson/Makefile.in	2022-01-24 01:57:49.579337878 +0000
@@ -134,7 +134,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/json.Po ./$(DEPDIR)/json_parser.Po \
diff -ur a/src/libs/zbxlog/Makefile.in b/src/libs/zbxlog/Makefile.in
--- a/src/libs/zbxlog/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxlog/Makefile.in	2022-01-24 01:46:37.341493697 +0000
@@ -133,7 +133,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/log.Po
diff -ur a/src/libs/zbxmodules/Makefile.in b/src/libs/zbxmodules/Makefile.in
--- a/src/libs/zbxmodules/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxmodules/Makefile.in	2022-01-24 01:58:35.115874887 +0000
@@ -133,7 +133,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/modules.Po
diff -ur a/src/libs/zbxnix/Makefile.in b/src/libs/zbxnix/Makefile.in
--- a/src/libs/zbxnix/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxnix/Makefile.in	2022-01-24 01:47:53.574633630 +0000
@@ -135,7 +135,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/control.Po ./$(DEPDIR)/coredump.Po \
diff -ur a/src/libs/zbxregexp/Makefile.in b/src/libs/zbxregexp/Makefile.in
--- a/src/libs/zbxregexp/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxregexp/Makefile.in	2022-01-24 01:59:00.452470079 +0000
@@ -133,7 +133,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/zbxregexp.Po
diff -ur a/src/libs/zbxsys/Makefile.in b/src/libs/zbxsys/Makefile.in
--- a/src/libs/zbxsys/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxsys/Makefile.in	2022-01-24 01:56:47.006185763 +0000
@@ -133,7 +133,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/mutexs.Po ./$(DEPDIR)/threads.Po
diff -ur a/src/libs/zbxsysinfo/agent/Makefile.in b/src/libs/zbxsysinfo/agent/Makefile.in
--- a/src/libs/zbxsysinfo/agent/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxsysinfo/agent/Makefile.in	2022-01-24 01:52:37.160300794 +0000
@@ -134,7 +134,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/libagentsysinfo_a-agent.Po \
diff -ur a/src/libs/zbxsysinfo/common/Makefile.in b/src/libs/zbxsysinfo/common/Makefile.in
--- a/src/libs/zbxsysinfo/common/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxsysinfo/common/Makefile.in	2022-01-24 01:54:35.986581026 +0000
@@ -150,7 +150,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/http.Po ./$(DEPDIR)/http_metrics.Po \
diff -ur a/src/libs/zbxsysinfo/linux/Makefile.in b/src/libs/zbxsysinfo/linux/Makefile.in
--- a/src/libs/zbxsysinfo/linux/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxsysinfo/linux/Makefile.in	2022-01-24 01:56:15.229612767 +0000
@@ -152,7 +152,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/hostname.Po \
diff -ur a/src/libs/zbxsysinfo/Makefile.in b/src/libs/zbxsysinfo/Makefile.in
--- a/src/libs/zbxsysinfo/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxsysinfo/Makefile.in	2022-01-24 01:52:24.480344209 +0000
@@ -149,7 +149,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/libzbxagent2sysinfo_a-sysinfo.Po \
diff -ur a/src/libs/zbxsysinfo/simple/Makefile.in b/src/libs/zbxsysinfo/simple/Makefile.in
--- a/src/libs/zbxsysinfo/simple/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/libs/zbxsysinfo/simple/Makefile.in	2022-01-24 01:55:44.889703404 +0000
@@ -134,7 +134,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/libsimplesysinfo_a-ntp.Po \
diff -ur a/src/zabbix_agent/logfiles/Makefile.in b/src/zabbix_agent/logfiles/Makefile.in
--- a/src/zabbix_agent/logfiles/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/zabbix_agent/logfiles/Makefile.in	2022-01-24 02:00:39.768860315 +0000
@@ -134,7 +134,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/logfiles.Po \
diff -ur a/src/zabbix_agent/Makefile.in b/src/zabbix_agent/Makefile.in
--- a/src/zabbix_agent/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/zabbix_agent/Makefile.in	2022-01-24 02:05:16.654776120 +0000
@@ -173,7 +173,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/libzbxagent_a-active.Po \
@@ -1124,7 +1124,7 @@
 install-data-hook:
 	$(MKDIR_P) "$(DESTDIR)$(AGENT_CONFIG_FILE).d"
 	$(MKDIR_P) "$(DESTDIR)$(LOAD_MODULE_PATH)"
-	test -f "$(DESTDIR)$(AGENT_CONFIG_FILE)" || cp "../../conf/zabbix_agentd.conf" "$(DESTDIR)$(AGENT_CONFIG_FILE)"
+	test -f "$(DESTDIR)$(AGENT_CONFIG_FILE)" || cp "../../../conf/zabbix_agentd.conf" "$(DESTDIR)$(AGENT_CONFIG_FILE)"
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -ur a/src/zabbix_get/Makefile.in b/src/zabbix_get/Makefile.in
--- a/src/zabbix_get/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/zabbix_get/Makefile.in	2022-01-24 02:01:37.648701438 +0000
@@ -140,7 +140,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/zabbix_get.Po
diff -ur a/src/zabbix_sender/Makefile.in b/src/zabbix_sender/Makefile.in
--- a/src/zabbix_sender/Makefile.in	2021-12-23 08:49:12.000000000 +0000
+++ b/src/zabbix_sender/Makefile.in	2022-01-24 02:02:01.105304085 +0000
@@ -143,7 +143,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/zabbix_sender.Po
