diff -ur a/src/libs/zbxaudit/Makefile.in b/src/libs/zbxaudit/Makefile.in
--- a/src/libs/zbxaudit/Makefile.in	2022-01-11 09:09:03.000000000 +0000
+++ b/src/libs/zbxaudit/Makefile.in	2022-01-24 12:18:48.778659229 +0000
@@ -136,7 +136,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/audit.Po ./$(DEPDIR)/audit_graph.Po \
diff -ur a/src/libs/zbxrtc/Makefile.in b/src/libs/zbxrtc/Makefile.in
--- a/src/libs/zbxrtc/Makefile.in	2022-01-11 09:09:03.000000000 +0000
+++ b/src/libs/zbxrtc/Makefile.in	2022-01-24 12:19:33.031920695 +0000
@@ -146,7 +146,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/rtc.Po ./$(DEPDIR)/rtc_client.Po \
diff -ur a/src/zabbix_server/service/Makefile.in b/src/zabbix_server/service/Makefile.in
--- a/src/zabbix_server/service/Makefile.in	2022-01-11 09:09:05.000000000 +0000
+++ b/src/zabbix_server/service/Makefile.in	2022-01-24 12:20:34.915138231 +0000
@@ -134,7 +134,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_builddir)/../include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/service_actions.Po \
