--- a/src/zabbix_proxy/Makefile.am	2022-07-25 08:48:49.824898308 +0000
+++ b/src/zabbix_proxy/Makefile.am	2022-08-13 12:12:01.323143329 +0000
@@ -127,4 +127,4 @@
 	$(MKDIR_P) "$(DESTDIR)$(PROXY_CONFIG_FILE).d"
 	$(MKDIR_P) "$(DESTDIR)$(EXTERNAL_SCRIPTS_PATH)"
 	$(MKDIR_P) "$(DESTDIR)$(LOAD_MODULE_PATH)"
-	test -f "$(DESTDIR)$(PROXY_CONFIG_FILE)" || cp "../../conf/zabbix_proxy.conf" "$(DESTDIR)$(PROXY_CONFIG_FILE)"
+	test -f "$(DESTDIR)$(PROXY_CONFIG_FILE)" || cp "$(top_srcdir)/conf/zabbix_proxy.conf" "$(DESTDIR)$(PROXY_CONFIG_FILE)"
--- a/src/zabbix_proxy/Makefile.in	2022-08-13 11:48:27.694381340 +0000
+++ b/src/zabbix_proxy/Makefile.in	2022-08-13 12:11:11.043435512 +0000
@@ -1095,7 +1095,7 @@
 	$(MKDIR_P) "$(DESTDIR)$(PROXY_CONFIG_FILE).d"
 	$(MKDIR_P) "$(DESTDIR)$(EXTERNAL_SCRIPTS_PATH)"
 	$(MKDIR_P) "$(DESTDIR)$(LOAD_MODULE_PATH)"
-	test -f "$(DESTDIR)$(PROXY_CONFIG_FILE)" || cp "../../conf/zabbix_proxy.conf" "$(DESTDIR)$(PROXY_CONFIG_FILE)"
+	test -f "$(DESTDIR)$(PROXY_CONFIG_FILE)" || cp "$(top_srcdir)/conf/zabbix_proxy.conf" "$(DESTDIR)$(PROXY_CONFIG_FILE)"
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
