--- a/configure	2022-07-25 08:48:51.224917239 +0000
+++ b/configure	2022-08-13 11:15:38.160147194 +0000
@@ -3451,7 +3451,7 @@
 ac_config_headers="$ac_config_headers include/common/config.h"
 
 
-DEFAULT_INCLUDES='-I$(top_srcdir)/include/common -I$(top_srcdir)/include'
+DEFAULT_INCLUDES='-I$(top_srcdir)/include/common -I$(top_srcdir)/include -I$(top_builddir)/include/common'
 
 
 # Make sure we can run config.sub.
--- a/configure.ac	2022-07-25 08:48:49.636895766 +0000
+++ b/configure.ac	2022-08-13 11:12:23.183734818 +0000
@@ -29,7 +29,7 @@
 
 AM_CONFIG_HEADER(include/common/config.h)
 
-AC_SUBST(DEFAULT_INCLUDES, ['-I$(top_srcdir)/include/common -I$(top_srcdir)/include'])
+AC_SUBST(DEFAULT_INCLUDES, ['-I$(top_srcdir)/include/common -I$(top_srcdir)/include -I$(top_builddir)/include/common'])
 
 AC_CANONICAL_HOST
 
