From 53c270da572d77f4c93c5f9f7d564bd0459db481 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Thu, 04 Jul 2013 22:47:03 +0300
Subject: [PATCH] ipv6 disabled by default

commit 707a0d73d7231b1821072712a7771c7aef140e21
Author: Stefan Saraev <stefan@saraev.ca>
Date:   Tue Jul 23 11:28:10 2013 +0300

    ipv6 disabled by default

---
diff --git a/src/ipconfig.c b/src/ipconfig.c
index fbeff8f..3eb61c4 100644
--- a/src/ipconfig.c
+++ b/src/ipconfig.c
@@ -1243,10 +1243,7 @@ static struct connman_ipconfig *create_ipv6config(int index)
 	ipv6config->index = index;
 	ipv6config->type = CONNMAN_IPCONFIG_TYPE_IPV6;
 
-	if (!is_ipv6_supported)
-		ipv6config->method = CONNMAN_IPCONFIG_METHOD_OFF;
-	else
-		ipv6config->method = CONNMAN_IPCONFIG_METHOD_AUTO;
+	ipv6config->method = CONNMAN_IPCONFIG_METHOD_OFF;
 
 	ipdevice = g_hash_table_lookup(ipdevice_hash, GINT_TO_POINTER(index));
 	if (ipdevice)
