--- a/CMakeLists.txt	2025-09-16 10:39:39.303923172 +0000
+++ b/CMakeLists.txt	2025-09-16 11:04:37.626398474 +0000
@@ -493,7 +493,7 @@
     add_test(NAME testrecurse COMMAND testrecurse WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
 endif()
 
-if(LIBXML2_WITH_DOCS OR LIBXML2_WITH_PYTHON)
+if(LIBXML2_WITH_DOCS)
     set(DOXYFILE ${CMAKE_CURRENT_SOURCE_DIR}/doc/Doxyfile)
     add_custom_command(
         OUTPUT doc/html doc/xml
@@ -519,7 +519,6 @@
             ${CMAKE_CURRENT_BINARY_DIR}
         DEPENDS
             python/generator.py
-            Doxygen
     )
 
     Python3_add_library(
